tests/components/matter/test_binary_sensor.py

```diff
@@ -13,6 +17,18 @@
 )
 
 
+@pytest.fixture(autouse=True)
+def binary_sensor_platform():
+    """Load only the binary sensor platform."""
+    with patch(
+        "homeassistant.components.matter.discovery.DISCOVERY_SCHEMAS",
```

Comment:
I had to patch this to make sure the event callback is for the binary sensor entity as the lock fixture creates more than one entity otherwise.